service.rb

```diff
@@ -27,6 +27,26 @@ def initialize(port)
         end
         response_message << "</ul>\n"
         response_message << daily_data_form
+      when ["POST", "/add-daily-data"]
+        content_type = "text/html"
+
+        while not client.eof?
```

Comment:
I know it was me who suggested [`IO#eof?`](https://rubydoc.info/stdlib/core/IO:eof%3F) in the first place but, for the reasons we discussed on Slack, I don’t think it’s doing anything useful here and arguably makes the code harder to understand. You might as well say `while true` or `loop do … end` instead. (Beware that the latter introduces a new local variable scope, so you’d need to initialise `content_length` outside of the block first so that it remains visible afterwards.)

Reply:
Ah TIL `loop` introduces a new variable scope. Switched to `while true`.

Reply:
Yeah, it’s cos [`Kernel#loop`](https://rubydoc.info/stdlib/core/Kernel:loop) is a regular method which takes a block (hence the `do`), unlike `while` which is a built-in language feature.